File: EffectHouseTutorials/skinSegmentation/Graph/ScriptTemplate/OnUpdate/CGOnUpdate.lua
local CGOnUpdate = CGOnUpdate or {}
CGOnUpdate.__index = CGOnUpdate

function CGOnUpdate.new()
    local self = setmetatable({}, CGOnUpdate)
    self.outputs = {}
    self.nexts = {}
    return self
end

function CGOnUpdate:setNext(index, func)
    self.nexts[index] = func
end

function CGOnUpdate:getOutput(index)
    return self.outputs[index]
end

function CGOnUpdate:update(sys, deltaTime)
    self.outputs[1] = deltaTime
    if self.nexts[0] then
        self.nexts[0]()
    end
end

return CGOnUpdate

File: EffectHouseTutorials/skinSegmentation/Graph/ScriptTemplate/OnUpdate/CGOnUpdate.js
/**
 * @file CGOnUpdate.js
 * @author
 * @date 2021/8/17
 * @brief CGOnUpdate.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGOnUpdate extends BaseNode {
  constructor() {
    super();
  }

  onUpdate(sys, deltatime) {
    this.outputs[1] = deltatime;

    if (this.nexts[0] !== undefined) {
      this.nexts[0]();
    }
  }
}

exports.CGOnUpdate = CGOnUpdate;
